src/Npgsql/Compatibility/CrateDB/CrateDBTimestampHandler.cs

```diff
@@ -0,0 +1,41 @@
+﻿using Npgsql.BackendMessages;
+using Npgsql.TypeHandling;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Npgsql.Compatibility.CrateDB
+{
+    class CrateDBTimestampHandlerFactory : NpgsqlTypeHandlerFactory<DateTime>
+    {
+        protected override NpgsqlTypeHandler<DateTime> Create(NpgsqlConnection conn) => new CrateDBTimestampHandler();
+    }
+
+    class CrateDBTimestampHandler : NpgsqlSimpleTypeHandler<DateTime>
+    {
+        internal static readonly DateTime PostgresEpoch = new DateTime(2000, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc);
+        
+        public override DateTime Read(NpgsqlReadBuffer buf, int len, FieldDescription fieldDescription = null)
+        {
+            var value = buf.ReadDouble();
+            try
+            {
+                return PostgresEpoch.AddSeconds(value);
```

Comment:
Unless I'm mistaken, this floating point format isn't specific to CrateDB at all - it's simply the old (and deprecated) PostgreSQL timestamp format (see note in https://www.postgresql.org/docs/current/static/datatype-datetime.html). If this is so, then it's possible to detect this via the standard `integer_datetimes` (see https://www.postgresql.org/docs/current/static/runtime-config-preset.html), and the built-in timestamp handler actually does this already (but throws a NotSupportedException). Try running `SHOW integer_datetimes`, if the result is false (or the parameter doesn't exist), then it's standard PostgreSQL rather than CrateDB-specific.

If I've correctly understood the situation, then please submit a separate PR that adds floating point format support to TimestampHandler, regardless of CrateDB.